packages/components-core/src/components/RankingCard/index.js

```diff
@@ -58,44 +58,71 @@ CitationWrapper.propTypes = {
 
 const InfoLayerWrapper = ({ imageSize, body, heading, readMoreLink }) => {
   const [open, setOpen] = useState(false);
+  const uniqueIdRef = useRef(
+    `info-layer-${Math.floor(Math.random() * 100000)}`
```

Comment:
Remove math.random 
checkout https://react.dev/reference/react/useId#usage
Do we need a ref or will aria control work with just ID?

Reply:
OK, leave a comment to replace to useId when we get to v18. but you still may not need `useRef`

Reply:
Yeah, that matches my understanding of when that feature is available. Also agree with the comment for future improvement. Good stuff.